### ISSUE TITLE
Write SciUnit validation tests for muscle_model

### ISSUE THREAD

**Issue by travs** [openworm/muscle_model#38]
This issue can be closed when two tests of connectivity are implemented and incorporated into the test suite using SciUnit

**Comment by gsarma** [comment]
Can we re-open specific issues with the two tests that you described above?  Just so that the issue names detail precisely what needs to be done.

**Comment by rgerkin** [comment]
@VahidGh @pgleeson Is the data [here](https://github.com/openworm/muscle_model/tree/master/BoyleCohen2008/data) the experimental data from Fig 3A,B or the simulated data from Fig 3C,D?

**Comment by rgerkin** [comment]
I've started this [here](https://github.com/openworm/muscle_model/blob/master/tests/SCU_BoyleCohen.ipynb).  

The first part looks more or less like what we have in ChannelWorm, except instead of loading it from the database I am loading it from the .csv files in the BoyleCohen2008/data directory.  The same scale issue is apparent as in the [corresponding ChannelWorm notebook](https://github.com/openworm/ChannelWorm/blob/master/tests/SCU_IVCurve_Model.ipynb).

The second part looks way off, and I'm not sure why.  Unless I'm doing something wrong, the K channel model has rectification properties that it shouldn't have (in addition to the scale issue, which is more trivial).  Possible things I could be doing wrong: simulating the wrong model somehow, computing currents at the wrong time point, ?  

The third part is the beginning of a whole cell test, including voltage and calcium time courses, but I don't know what data to use to write a test for this.

**Comment by VahidGh** [comment]
@rgerkin, yes, in addition to [this](https://github.com/openworm/muscle_model/tree/master/BoyleCohen2008/MatlabSupport/Main_Version/data), in which you can find needed parameters like C_m, dense conductance, and e_rev for each channel.

**Comment by rgerkin** [comment]
@VahidGh "Yes" it is experimental data or "Yes" it is simulated data?  Thanks for the link.

**Comment by VahidGh** [comment]
@rgerkin, The link in your comment, is experimental data. And the link addressed by me is simulated data.

**Comment by pgleeson** [comment]
The data here: https://github.com/openworm/muscle_model/tree/master/BoyleCohen2008/data seems like experimental data, but no info/notes associated with it. 

I'd suggest first making sure their data matches the output of their matlab scripts before looking at the match to nml2 version. You could probably work with Rayner's python version too.

Note, the ca trace should work for the nml2 version (see https://github.com/openworm/ChannelWorm/issues/96#issuecomment-134233138), but note the K iv curve is for k_fast+k_slow, and there is no current way to get the total K current from the nml2 version (unless making 2 ivCurve analysis scripts and adding them)

**Comment by slarson** [comment]
Related issue: https://github.com/openworm/muscle_model/issues/64

**Comment by slarson** [comment]
Revisiting this today in the context of the scientific roadmap.  Here are the steps that are required to close this off:

- [ ] Make a README.md for the data we find [here](https://github.com/openworm/muscle_model/tree/master/BoyleCohen2008/data) so we have references for the data that the model will be constrained again
- [ ] (Optional) Incorporate this data so it can be sourced from PyOpenWorm (cc: @mwatts15 )
- [ ] Using [this notebook](https://github.com/openworm/muscle_model/blob/master/tests/SCU_BoyleCohen.ipynb) as a starting point, incorporate the data above.
- [ ] Also incorporate the methods for optimizing the channels so the figures have good overlapping matches from the [channelworm fitter](https://github.com/openworm/ChannelWorm/blob/master/channelworm/fitter/examples/EGL-19-2.py) for EGL-19 (cc: @VahidGh)
- [ ] (Optional) Incorporate [the channel data](https://github.com/openworm/ChannelWorm/tree/master/channelworm/fitter/examples/egl-19-data) from ChannelWorm into PyOpenWorm (cc: @mwatts15 )
- [ ] Update .travis.yml for this project to execute the test from the notebook via SciUnit (cc: @rgerkin )

**Comment by slarson** [comment]
We need to sort out the topic of where the output objects from the notebooks will go once travis generates them.  Previous they had been going into s3 but this may not be the best strategy going forward.  (cc @rgerkin )
